### PR TITLE
drivers: serial: ns16550: Moved PCIe probe to init function

### PR DESCRIPTION
**Pull request by akanisetti**
Moved PCIe probe from configure function to init function
because whenever uart_configure api is called MMIO address
is getting updated.

Signed-off-by: Anisetti Avinash Krishna <anisetti.avinash.krishna@intel.com>

### CODE REVIEW

File: drivers/serial/uart_ns16550.c [hunk 1]
```diff
@@ -332,38 +332,12 @@ static int uart_ns16550_configure(const struct device *dev,
 
 	k_spinlock_key_t key = k_spin_lock(&dev_data->lock);
 
-	ARG_UNUSED(dev_data);
-	ARG_UNUSED(dev_cfg);
```
Comment (finikorg) [inline drivers/serial/uart_ns16550.c:336]: I think the change should not affect `ARG_UNUSED()`

Comment (akanisetti) [reply drivers/serial/uart_ns16550.c:336]: > I think the change should not affect `ARG_UNUSED()`

dev_data and dev_cfg are used without any conditions so I thought ARG_UNUSED() is not needed.

File: drivers/serial/uart_ns16550.c [hunk 2]
```diff
@@ -950,12 +950,21 @@ static int uart_ns16550_line_ctrl_set(const struct device *dev,
...
 	struct uart_ns16550_dev_data *data = dev->data;
-	uint32_t mdc, chg;
+	const struct uart_ns16550_device_config *const dev_cfg = dev->config;
+	uint32_t mdc, chg, pclk = 0;
 	k_spinlock_key_t key;
 
+	if (dev_cfg->sys_clk_freq != 0U) {
+		pclk = dev_cfg->sys_clk_freq;
+	} else {
+		if (device_is_ready(dev_cfg->clock_dev)) {
```
Comment (finikorg) [inline drivers/serial/uart_ns16550.c:956]: do we need `device_is_ready()` check here?

Comment (akanisetti) [reply drivers/serial/uart_ns16550.c:956]: > do we need `device_is_ready()` check here?

It is being checked in configuration api so added it here also.

File: drivers/serial/uart_ns16550.c [hunk 3]
```diff
@@ -950,12 +950,21 @@ static int uart_ns16550_line_ctrl_set(const struct device *dev,
 				      uint32_t ctrl, uint32_t val)
 {
 	struct uart_ns16550_dev_data *data = dev->data;
-	uint32_t mdc, chg;
+	const struct uart_ns16550_device_config *const dev_cfg = dev->config;
+	uint32_t mdc, chg, pclk = 0;
```
Comment (finikorg) [inline drivers/serial/uart_ns16550.c:950]: If you start using `0U` below, probably you need to use it also here?

Comment (akanisetti) [reply drivers/serial/uart_ns16550.c]: Updated

**Review by jhedberg — APPROVED** [review]

**Review by finikorg — APPROVED** [review]

**Review by dcpleung — APPROVED** [review]

**Pull request merged**